docs/index.md

```diff
@@ -1128,6 +1128,44 @@ app/components
 </div>
 ```
 
+##### Translations (experimental)
+
+To use experimental support `I18n` translations, include `ViewComponent::SidecarI18n`:
+
+`app/components/example_component.rb`
+
+```ruby
+module ExampleComponent < ApplicationComponent
+  include ViewComponent::SidecarI18n
+end
+```
+
+Then add a sidecar YAML file with translations for the component:
+
+`app/components/example_component.yml`
+
+```yml
+en:
+  hello: "Hello world!"
+```
+
+The translations will then be available as usual:
+
+`app/components/example_component.html.erb`
+
+```erb
+<p><%= t("hello") %></p>
+```
+
+To access global translations, explicitly mention `helper` or `I18n`:
```

Comment:
Would it be possible to have `t` also access global translations? I think it's a little awkward to re-define `t` with a local scope and then expect consumers to know when to use the `helpers` prefix.